File: CS/WindowsFormsApplication202/ImageSliderHelper.cs
using DevExpress.Utils;
using DevExpress.XtraEditors.CustomEditor;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace WindowsFormsApplication202 {
    static public class ImageSliderHelper {
        static int currentImageIndex = 0;
        static ImageCollection currentCollection = null;
        static GridView view = null;
        static ImageCollection[] images = null;

        public static void BindImageSlider(GridControl grid, string columnFieldName, params ImageCollection[] rowImages) {
            images = rowImages;
            RepositoryItemAnyControl repository = new RepositoryItemAnyControl();
            grid.RepositoryItems.Add(repository);
            MyImageSlider slider = new MyImageSlider();
            view = grid.MainView as GridView;
            repository.Control = slider;
            view.Columns["Image"].ColumnEdit = repository;
            view.Columns["Description"].OptionsColumn.AllowEdit = false;
            view.RowHeight = slider.Size.Height;
            view.ShownEditor += view_ShownEditor;
        }

        private static void view_ShownEditor(object sender, EventArgs e) {
            GridView view = sender as GridView;
            AnyControlEdit edit = view.ActiveEditor as AnyControlEdit;
            if (edit != null) {
                currentCollection = images[view.FocusedRowHandle];
                MyImageSlider slider = edit.Properties.Control as MyImageSlider;
                slider.VirtualMode = true;
                slider.GetImage += slider_GetImage;
                slider.CanGetNextPrevImage += slider_CanGetNextPrevImage;
                slider.UpdateImage();
                currentImageIndex = 0;
            }
        }

        static void slider_CanGetNextPrevImage(object sender, DevExpress.XtraEditors.Controls.CanGetNextPrevImageEventArgs e) {
            if (e.IsNext) e.CanGetImage = (currentImageIndex < currentCollection.Images.Count - 1);
            else e.CanGetImage = (currentImageIndex > 0);
        }

        static void slider_GetImage(object sender, DevExpress.XtraEditors.Controls.GetImageEventArgs e) {
            if (currentCollection != null) {
                if (e.IsStartUp) {
                    e.Image = currentCollection.Images[0];
                    return;
                }
                if (e.IsNext) {
                    e.Image = currentCollection.Images[currentImageIndex + 1];
                    if (e.CurrentImageUpdated) ++currentImageIndex;
                }
                else if (e.IsPrev) {
                    e.Image = currentCollection.Images[currentImageIndex - 1];
                    if (e.CurrentImageUpdated) --currentImageIndex;
                }
            }
        }
    }
}

File: VB/WindowsFormsApplication202/ImageSliderHelper.vb
Imports DevExpress.Utils
Imports DevExpress.XtraEditors.CustomEditor
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace WindowsFormsApplication202
	Public NotInheritable Class ImageSliderHelper

		Private Sub New()
		End Sub

		Private Shared currentImageIndex As Integer = 0
		Private Shared currentCollection As ImageCollection = Nothing
		Private Shared view As GridView = Nothing
		Private Shared images() As ImageCollection = Nothing

		Public Shared Sub BindImageSlider(ByVal grid As GridControl, ByVal columnFieldName As String, ParamArray ByVal rowImages() As ImageCollection)
			images = rowImages
			Dim repository As New RepositoryItemAnyControl()
			grid.RepositoryItems.Add(repository)
			Dim slider As New MyImageSlider()
			view = TryCast(grid.MainView, GridView)
			repository.Control = slider
			view.Columns("Image").ColumnEdit = repository
			view.Columns("Description").OptionsColumn.AllowEdit = False
			view.RowHeight = slider.Size.Height
			AddHandler view.ShownEditor, AddressOf view_ShownEditor
		End Sub

		Private Shared Sub view_ShownEditor(ByVal sender As Object, ByVal e As EventArgs)
			Dim view As GridView = TryCast(sender, GridView)
			Dim edit As AnyControlEdit = TryCast(view.ActiveEditor, AnyControlEdit)
			If edit IsNot Nothing Then
				currentCollection = images(view.FocusedRowHandle)
				Dim slider As MyImageSlider = TryCast(edit.Properties.Control, MyImageSlider)
				slider.VirtualMode = True
				AddHandler slider.GetImage, AddressOf slider_GetImage
				AddHandler slider.CanGetNextPrevImage, AddressOf slider_CanGetNextPrevImage
				slider.UpdateImage()
				currentImageIndex = 0
			End If
		End Sub

		Private Shared Sub slider_CanGetNextPrevImage(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.CanGetNextPrevImageEventArgs)
			If e.IsNext Then
				e.CanGetImage = (currentImageIndex < currentCollection.Images.Count - 1)
			Else
				e.CanGetImage = (currentImageIndex > 0)
			End If
		End Sub

		Private Shared Sub slider_GetImage(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.GetImageEventArgs)
			If currentCollection IsNot Nothing Then
				If e.IsStartUp Then
					e.Image = currentCollection.Images(0)
					Return
				End If
				If e.IsNext Then
					e.Image = currentCollection.Images(currentImageIndex + 1)
					If e.CurrentImageUpdated Then
						currentImageIndex += 1
					End If
				ElseIf e.IsPrev Then
					e.Image = currentCollection.Images(currentImageIndex - 1)
					If e.CurrentImageUpdated Then
						currentImageIndex -= 1
					End If
				End If
			End If
		End Sub
	End Class
End Namespace
